app/assets/javascripts/discourse/app/lib/uploads.js

```diff
@@ -343,3 +343,14 @@ function displayErrorByResponseStatus(status, body, fileName, siteSettings) {
 
   return;
 }
+
+export function bindFileInputChangeListener(element, fileCallbackFn) {
+  function changeListener(event) {
+    const files = Array.from(event.target.files);
+    files.forEach((file) => {
+      fileCallbackFn(file);
+    });
+  }
+  element.addEventListener("change", changeListener);
```

Comment:
do we ever remove the listener here ?

Reply:
Yes, that's what these lines are doing:

https://github.com/discourse/discourse/blob/f7ed34f1e0351926af53f08b96ff33eb529b1fd2/app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js#L368-L373

https://github.com/discourse/discourse/blob/f7ed34f1e0351926af53f08b96ff33eb529b1fd2/app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js#L54-L57